ast/Trees.h

```diff
@@ -899,6 +899,65 @@ EXPRESSION(Send) {
     // Remove all arguments to the function, including the block argument.
     void clearArgs();
 
+    template <typename PtrT> struct KwPair {
+        absl::Span<PtrT> span;
+
+        PtrT &key() {
+            return span[0];
+        }
+
+        PtrT &value() {
+            return span[1];
+        }
+    };
+
+    // We're going to use this type to encapsulate both the span over the kwargs,
+    // and also as the iterator type over that selfsame span.
+    template <typename PtrT> struct KwPairSpan {
+        absl::Span<PtrT> span;
+
+        KwPairSpan begin() const {
+            return KwPairSpan{span};
+        }
+
+        KwPairSpan end() const {
+            return KwPairSpan{absl::MakeSpan(span.end(), span.end())};
+        }
```

Comment:
This is a nice way to describe the end of the range of pairs! I always get stuck thinking about how to encode the end of a c++ iterator.

Would it be worth adding an `ENFORCE` here and on `begin` that the `size` of the span is even?